engine/pkg/orm/client.go

```diff
@@ -587,35 +589,34 @@ func (c *metaOpsClient) QueryResourcesByJobID(ctx context.Context, jobID string)
 	return resources, nil
 }
 
-// QueryResourcesByExecutorID query all resources of the executor_id
-func (c *metaOpsClient) QueryResourcesByExecutorID(ctx context.Context, executorID string) ([]*resModel.ResourceMeta, error) {
+// QueryResourcesByExecutorIDs query all resources of the executorIDs
+func (c *metaOpsClient) QueryResourcesByExecutorIDs(
+	ctx context.Context, executorIDs ...engineModel.ExecutorID,
+) ([]*resModel.ResourceMeta, error) {
 	var resources []*resModel.ResourceMeta
 	if err := c.db.WithContext(ctx).
-		Where("executor_id = ?", executorID).
+		Where("executor_id in ?", executorIDs).
 		Find(&resources).Error; err != nil {
 		return nil, errors.ErrMetaOpFail.Wrap(err)
 	}
 
 	return resources, nil
 }
 
-// DeleteResourcesByExecutorID delete all the resources of executorID
-func (c *metaOpsClient) DeleteResourcesByExecutorID(ctx context.Context, executorID engineModel.ExecutorID) (Result, error) {
-	result := c.db.WithContext(ctx).
-		Where("executor_id = ?", executorID).
-		Delete(&resModel.ResourceMeta{})
-	if result.Error == nil {
-		return &ormResult{rowsAffected: result.RowsAffected}, nil
+// DeleteResourcesByTypeAndExecutorIDs delete a specific type of resources of executorID
+func (c *metaOpsClient) DeleteResourcesByTypeAndExecutorIDs(
+	ctx context.Context, resType resModel.ResourceType, executorIDs ...engineModel.ExecutorID,
+) (Result, error) {
+	var result *gorm.DB
+	if len(executorIDs) == 1 {
+		result = c.db.WithContext(ctx).
+			Where("executor_id = ? and id like ?", executorIDs[0], fmt.Sprintf("/%s%%", resType)).
```

Comment:
Prefer adding `fmt.Sprintf("/%s%%", resType)` format to `pkg/externalresource/model/model.go`.

Reply:
Done, PTAL